### PR TITLE
Add what's new in TiDB 4.0

### PR DESCRIPTION
**Pull request by TomShawn**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3537
- Other reference link(s):

### CODE REVIEW

File: whats-new-in-tidb-4.0.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+- Support using the Index Merge feature to access tables. When you make a query on a single table, the TiDB optimizer automatically reads multiple index data according to the query condition and makes a union of the result, which improves the performance of querying on a single table. See [Index Merge](/index-merge.md) for details.
+- Support the expression index feature (**experimental**). The expression index is also called the function-based index. When you create an index, the index fields do not have to be a specific column but can be an expression calculated from one or more columns. This feature is useful for quickly accessing the calculation-based tables. See [Expression index](/sql-statements/sql-statement-create-index.md) for details.
+- Support AutoRandom keys as an extended syntax for the TiDB columnar attribute (**experimental**). AutoRandom is designed to address the hotspot issue caused by the auto-increment column and provides a low-cost migration solution from MySQL for users who work with auto-increment columns. See [AutoRandom Key](/auto-random.md) for details.
```
Comment (ran-huang) [inline whats-new-in-tidb-4.0.md:61]: The linked doc uses `AUTO_RANDOM`.
```suggestion
- Support `AUTO_RANDOM` keys as an extended syntax for the TiDB columnar attribute (**experimental**). `AUTO_RANDOM` is designed to address the hotspot issue caused by the auto-increment column and provides a low-cost migration solution from MySQL for users who work with auto-increment columns. See [`AUTO_RANDOM` Key](/auto-random.md) for details.
```

File: whats-new-in-tidb-4.0.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+- Add system tables that provide information of cluster topology, configuration, logs, hardware, operating systems, and slow queries, which helps DBAs to quickly learn, analyze system metrics. See [SQL Diagnosis](/system-tables/system-table-sql-diagnosis.md) for details.
+
+    - Add system tables that provide information of cluster topology, configuration, logs, hardware, operating systems to help DBAs quickly learn the cluster configuration and status:
+        - The `cluster_info` table that stores the cluster topology information.
+        - The `cluster_log` table that stores log information.
+        - The `cluster_hardware` and `cluster_systeminfo` tables that save the server information of hardware and operating system.
+    - Add information tables that provide information of slow queries, diagnostic results, and performance monitoring to help DBAs quickly analyze the system bottleneck:
+        - The `cluster_slow_query` table that records the global slow query information.
+        - The `cluster_proacesslist` table that records the global processlist.
```
Comment (ran-huang) [inline whats-new-in-tidb-4.0.md:70]:
```suggestion
        - The `cluster_processlist` table that records the global processlist.
```

File: whats-new-in-tidb-4.0.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+- Benchmark test using TiUP Benchmark, which enables you to easily deploy the benchmark performance test tool, and provides workload for TPC-C and TPC-H tests.
+
+### Transaction
+
+- The pessimistic transaction is now provided for general availability as the default transaction mode. Support the Read Committed isolation level and the `SELECT FOR UPDATE NOWAIT` syntax. See [Pessimistic Transaction Model](/pessimistic-transaction.md) for details.
+- Support large transactions. Increase the upper limit on transaction size from 10 MB to 10 GB. Support both the pessimistic transaction and optimistic transaction.
+
+### SQL features
+
+- Introduce the automatic capture and evolution of SQL Bind to SQL Plan Management, which improves the usability and stability of the execution plan. See [SQL Plan Management](/execution-plan-binding.md) for details.
```
Comment (TomShawn) [inline whats-new-in-tidb-4.0.md:52]:
```suggestion
- Introduce the automatic capture and evolution of SQL Bind to SQL Plan Management, which improves the usability and stability of the execution plan. See [SQL Plan Management](/sql-plan-management.md) for details.
```

**Pull request merged**